### PR TITLE
MTDSA-6273 service and connector for amend disclosures

### PR DESCRIPTION
**Pull request by yangwong9664**

### CODE REVIEW

File: app/v1/connectors/AmendDisclosuresConnector.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                                          val appConfig: AppConfig) extends BaseDesConnector {
+
+  def amendDisclosures(request: AmendDisclosuresRequest)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[DesOutcome[Unit]] = {
+
+    import v1.connectors.httpparsers.StandardDesHttpParser._
+
+    put(
+      uri = DesUri[Unit](s"some-placeholder/disclosures/${request.nino}/${request.desTaxYear.value}"), body = request.body
```
Comment (Taiwo-AB) [inline app/v1/connectors/AmendDisclosuresConnector.scala:37]: Wrong uri

File: test/v1/connectors/AmendDisclosuresConnectorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  "AmendDisclosuresConnector" when {
+    "amendDisclosures" must {
+      "return a 204 status for a success scenario" in new Test {
+        val outcome = Right(ResponseWrapper(correlationId, ()))
+
+        MockedHttpClient
+          .put(
+            url = s"$baseUrl/some-placeholder/disclosures/$nino/$taxYear",
```
Comment (Taiwo-AB) [inline test/v1/connectors/AmendDisclosuresConnectorSpec.scala:60]: wrong url

File: app/v1/connectors/AmendDisclosuresConnector.scala (hunk 1 above)
Comment (yangwong9664) [reply app/v1/connectors/AmendDisclosuresConnector.scala:37]: What is the uri? I can't see one in the Jira story.

File: test/v1/connectors/AmendDisclosuresConnectorSpec.scala (hunk 2 above)
Comment (yangwong9664) [reply test/v1/connectors/AmendDisclosuresConnectorSpec.scala:60]: What is the uri? I can't see one in the Jira story.

File: test/v1/services/AmendDisclosuresServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  private val nino = "AA112233A"
+  private val taxYear = "2019"
+  private val correlationId = "X-corr"
+
+  val amendTaxAvoidance: AmendTaxAvoidance = AmendTaxAvoidance("12345","2020-12")
+
+  val amendDisclosuresRequest: AmendDisclosuresRequest = AmendDisclosuresRequest(
+      nino = Nino(nino),
+      desTaxYear = DesTaxYear(taxYear),
```
Comment (Taiwo-AB) [inline test/v1/services/AmendDisclosuresServiceSpec.scala:39]: Can you please change desTaxYear field in the AmendDisclosuresRequest to taxYear and make appropriate changes in the connector, connector spec and so on.

File: app/v1/connectors/AmendDisclosuresConnector.scala (hunk 1 above)
Comment (Taiwo-AB) [reply app/v1/connectors/AmendDisclosuresConnector.scala:37]: It's s"disc-placeholder/disclosures/{request.nino}/${request.desTaxYear.value}"

File: test/v1/services/AmendDisclosuresServiceSpec.scala (hunk 3 above)
Comment (yangwong9664) [reply test/v1/services/AmendDisclosuresServiceSpec.scala:39]: Is there a particular reason, as income-api uses it too? Also that DesTaxYear is already in master, you merged it last week. But I can change it as part of this story if needed.

File: app/v1/connectors/AmendDisclosuresConnector.scala (hunk 1 above)
Comment (yangwong9664) [reply app/v1/connectors/AmendDisclosuresConnector.scala:37]: Thank you, do you have a reference to that? Might also be good to include those uri's in future tech specs

File: test/v1/services/AmendDisclosuresServiceSpec.scala (hunk 3 above)
Comment (Taiwo-AB) [reply test/v1/services/AmendDisclosuresServiceSpec.scala:39]: Not the DesTaxYear but the desTaxYear so I mean the field instead.

File: app/v1/connectors/AmendDisclosuresConnector.scala (hunk 1 above)
Comment (Taiwo-AB) [reply app/v1/connectors/AmendDisclosuresConnector.scala:37]: The stub should have it

Comment (yangwong9664) [reply app/v1/connectors/AmendDisclosuresConnector.scala:37]: Okay thanks, would be helpful to have it in the story and tech-spec. I'll add it.

File: test/v1/services/AmendDisclosuresServiceSpec.scala (hunk 3 above)
Comment (yangwong9664) [reply test/v1/services/AmendDisclosuresServiceSpec.scala:39]: Ah okay you mean the field name/case class parameter name, I'll make that change, thanks.

Comment (Taiwo-AB) [reply test/v1/services/AmendDisclosuresServiceSpec.scala:39]: Yep, thanks

File: test/v1/services/AmendDisclosuresServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+import scala.concurrent.Future
+
+class AmendDisclosuresServiceSpec extends ServiceSpec {
+
+  private val nino = "AA112233A"
+  private val taxYear = "2019"
+  private val correlationId = "X-corr"
+
+  val amendTaxAvoidance: AmendTaxAvoidance = AmendTaxAvoidance("12345","2020-12")
```
Comment (Taiwo-AB) [inline test/v1/services/AmendDisclosuresServiceSpec.scala:35]: Use the right values i.e. 14211123 and 2020-21

File: test/v1/connectors/AmendDisclosuresConnectorSpec.scala [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import v1.models.request.disclosures.{AmendDisclosuresRequest, AmendDisclosuresRequestBody, AmendTaxAvoidance}
+
+import scala.concurrent.Future
+
+class AmendDisclosuresConnectorSpec extends ConnectorSpec {
+
+  private val nino: String = "AA111111A"
+  private val taxYear: String = "2019"
+
+  val amendTaxAvoidance: AmendTaxAvoidance = AmendTaxAvoidance("12345","2020-12")
```
Comment (Taiwo-AB) [inline test/v1/connectors/AmendDisclosuresConnectorSpec.scala:33]: Use the right values i.e. 14211123 and 2020-21

File: test/v1/services/AmendDisclosuresServiceSpec.scala (hunk 4 above)
Comment (yangwong9664) [reply test/v1/services/AmendDisclosuresServiceSpec.scala:35]: Sure. Made that change for the test data.

File: test/v1/connectors/AmendDisclosuresConnectorSpec.scala (hunk 5 above)
Comment (yangwong9664) [reply test/v1/connectors/AmendDisclosuresConnectorSpec.scala:33]: Sure. Made that change for the test data.

**Pull request merged**